File: 3.Offstand/#.Source/OffStand/app/src/main/java/com/fundroid/offstand/MainActivity.java
package com.fundroid.offstand;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.fundroid.offstand.data.remote.ConnectionManager;

import java.net.InetAddress;
import java.net.UnknownHostException;

import androidx.appcompat.app.AppCompatActivity;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.reactivex.schedulers.Schedulers;

import static com.fundroid.offstand.core.AppConstant.ROOM_PORT;

public class MainActivity extends AppCompatActivity {
    static final String TAG = "[MAIN]";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 버터나이프 사용
        ButterKnife.bind(this);
    }

    @OnClick(R.id.button_room)
    public void goRoom() {
        Log.d(TAG, "goRoom");

        Intent intent = new Intent(MainActivity.this, RoomActivity.class);
        startActivity(intent);
        finish();
    }

    @OnClick(R.id.button_room2)
    public void goRoom2() {
        Log.d(TAG, "goRoom2");

        Intent intent = new Intent(MainActivity.this, Room2Activity.class);
        startActivity(intent);
        finish();
    }

    @OnClick(R.id.button_play)
    public void goPlay() {
        Log.d(TAG, "goPlay");

        Intent intent = new Intent(MainActivity.this, PlayActivity.class);
        startActivity(intent);
        finish();
    }

    @OnClick(R.id.button_server)
    public void createSocket() {
        ConnectionManager.createServerThread(ROOM_PORT, 5)
                .subscribeOn(Schedulers.io())
                .observeOn(Schedulers.io())
                .subscribe(message -> {
                    Log.d(TAG, "createSocket message " + message);
                }, onError -> {
                    Log.d(TAG, "createSocket onError " + onError.getMessage());
                });
    }

    @OnClick(R.id.button_client)
    public void connectSocket() {

        new Thread(() -> {
            try {
                ConnectionManager.createClientThread(InetAddress.getLocalHost(), ROOM_PORT)
                        .subscribeOn(Schedulers.io())
                        .observeOn(Schedulers.io())
                        .subscribe(onNext -> {
                            Log.d(TAG, "enterRoom thread " + Thread.currentThread().getName());
                            Log.d(TAG, "enterRoom onNext " + onNext);
                            //                            Toast.makeText(context, onNext, Toast.LENGTH_SHORT).show();
                        }, onError -> {
                            Log.d(TAG, "enterRoom onError " + onError.getMessage());
                        });
            } catch (UnknownHostException e) {
                Log.e("lsc","e " + e.getMessage());
            }
        }).start();



    }
}

File: 4. Routine/#.Private/DDang/RoomDB/app/src/main/java/com/roadwise/roomdb/MainActivity.kt
package com.roadwise.roomdb

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var catDb : CatDB? = null
    private var catList = listOf<Cat>()
    lateinit var mAdapter : CatAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        catDb = CatDB.getInstance(this)
        mAdapter = CatAdapter(this, catList)

        val r = Runnable {
            try {
                catList = catDb?.catDao()?.getAll()!!
                mAdapter.notifyDataSetChanged()

                mRecyclerView.adapter = mAdapter
                mRecyclerView.layoutManager = LinearLayoutManager(this)
                mRecyclerView.setHasFixedSize(true)
            } catch (e: Exception) {
                Log.d("tag", "Error - $e")
            }
        }

        val thread = Thread(r)
        thread.start()

        mAddBtn.setOnClickListener {
            val i = Intent(this, AddActivity::class.java)
            startActivity(i)
            finish()
        }
    }

    override fun onDestroy() {
        CatDB.destroyInstance()
        catDb = null
        super.onDestroy()
    }
}